lib/cocoapods/target.rb

```diff
@@ -33,6 +33,7 @@ class Target
     # @return [BuildSettings] the build settings for this target.
     #
     attr_reader :build_settings
+    alias build_settings_by_config build_settings
```

Comment:
this seems a bit odd, since `AggregateTarget` will return a `Hash` here and the name implies its a `Hash`, but the base `Target` class returns a single `BuildSettings` instance

Reply:
this was a good catch I am looking further to ensure whether this is a documentation issue or a actual data storage issue.

Reply:
@amorde you are right, the base class returns just `BuildSettings`whereas subclasses return a `Hash` will see if I can update.